velox/functions/prestosql/aggregates/tests/MapUnionSumTest.cpp

```diff
@@ -403,5 +403,40 @@ TEST_F(MapUnionSumTest, floatingPointKeys) {
   testAggregations({data}, {}, {"map_union_sum(c1)"}, {expected});
 }
 
+TEST_F(MapUnionSumTest, nanKeys) {
+  // Verify that NaNs with different binary representations are considered equal
+  // and deduplicated when used as keys in the output map.
+  constexpr double kNan = std::numeric_limits<double>::quiet_NaN();
+  constexpr double kSNaN = std::numeric_limits<double>::signaling_NaN();
+
+  // Global aggregation.
+  auto data = makeRowVector(
+      {makeNullableMapVector<double, int32_t>({
```

Comment:
nit: curious if we have double, double as map vector and NaN is a value , do we throw ?